src/components/CanvasWrapper.tsx

```diff
@@ -375,7 +375,7 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
         props.onTrackClicked(newTrack);
       }
     },
-    [canv, props.dataset]
+    [canv, props.dataset, props.onTrackClicked]
```

Comment:
Fixes a bug where this callback wasn't updating when the `onTrackClicked` changed. This caused annotation behavior not to work on click.